src/utilities/mergeParams.ts

```diff
@@ -1,12 +1,15 @@
 import { Param } from '@/types'
+import { asArray } from '@/utilities'
 
-export function mergeParams(...paramRecords: Record<string, Param[]>[]): Record<string, Param[]> {
+export function mergeParams(...paramRecords: Record<string, Param | Param[]>[]): Record<string, Param[]> {
   return paramRecords.reduce<Record<string, Param[]>>((combined, collection) => {
     Object.entries(collection).forEach(([key, params]) => {
+      const paramsArray = asArray(params)
+
       if (key in combined) {
-        params.forEach(param => combined[key].push(param))
+        paramsArray.forEach(param => combined[key].push(param))
       } else {
-        combined[key] = [...params]
+        combined[key] = [...paramsArray]
```

Comment:
I like most of this, TS doesn't like `combined[key] ??= []`, it's that annoying error we get when doing a reduce

> Unnecessary conditional, expected left-hand side of `??` operator to be possibly null or undefined

because we're saying the "combined" is `Record<string, Param[]>`, so it say's it won't be undefined. I think I'm going to do the following

```ts
if (!(key in combined)) {
  combined[key] = []
}

combined[key].push(...asArray(params))
```